src/mol-geo/geometry/marker-data.ts

```diff
@@ -11,18 +11,35 @@ import { TextureImage, createTextureImage } from '../../mol-gl/renderable/util';
 export type MarkerData = {
     tMarker: ValueCell<TextureImage<Uint8Array>>
     uMarkerTexDim: ValueCell<Vec2>
+    markerAverage: ValueCell<number>
+    markerStatus: ValueCell<number>
+}
+
+export function getMarkersAverage(array: Uint8Array, count: number): number {
+    if (count === 0) return 0;
+    let sum = 0;
+    for (let i = 0; i < count; ++i) {
+        if (array[i]) sum += 1;
+    }
+    return sum / count;
```

Comment:
We are only ever interested in empty / full / has-non-zero element.

Maybe this code would work better on average? (I haven't ran it so there might be some error, but the overall idea should be sound).

```ts
export function getMarkersState(array: Uint8Array, count: number): 0 | 0.5 | 1 {
    // 0 empty
    // 0.5 partially full
    // 1 full

    if (count === 0) return 0;

    let zero = +!array[0];
    for (let i = 1; i < count; ++i) {
        const v = array[i] && 1;
        // (v && zero) || (!v && !zero) .. I hope :)
        if (!(v ^ zero)) return 0.5;
        if (v) zero = 0;
    }
    return zero ? 0 : 1;
}
```

Reply:
Thanks, nice idea. I added some other code to handle cases where early return would be helpful. With that is seems to be faster to have a branch-less loop to calculate this.